### PR TITLE
Fgdc mappings

### LINKED ISSUE #18: Develop mapping for Harvard FGDC

### PR DESCRIPTION
**Pull request by cmh2166**
Closes #18 .

### CODE REVIEW

File: lib/traject/macros/xml.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def extract_xml(xpath, namespaces, options = {})
       end
     end
 
+    def extract_field(xml, xpath, namespaces)
```
Comment (cbeer) [inline lib/traject/macros/xml.rb]: Is this method used?

Comment (cmh2166) [inline lib/traject/macros/xml.rb]: is it expanded by the fgdc + mods xpath methods? i can check, but that's my understanding

File: lib/traject/fgdc_config.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+require_relative 'macros/xml'
+
+extend Macros::DLME
+extend Macros::Xml
+extend Macros::FGDC
+
+settings do
+  provide 'writer_class_name', 'DlmeJsonResourceWriter'
+  provide 'reader_class_name', 'XmlReader'
+  provide 'allow_empty_fields', true
```
Comment (jcoyne) [inline lib/traject/fgdc_config.rb]: We can remove this line.

File: lib/traject/fgdc_config.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  provide 'agg_provider', 'Harvard University Library'
```
Comment (jcoyne) [inline lib/traject/fgdc_config.rb]: Move these three lines into `config/settings.yml` (You may need to rebase first)

File: lib/traject/macros/fgdc.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    def harvard_uuid(record)
+      urls = record.xpath('/*/idinfo/citation/citeinfo/onlink', NS).map(&:text)
+      urls.first.split('CollName=')[1]
+    end
+
+    def extract_uuid(record, context)
+      institution = context.settings.fetch('agg_provider')
+      if institution.include?('Harvard')
+        harvard_uuid(record)
+      elsif institution.include?('MIT')
```
Comment (jcoyne) [inline lib/traject/macros/fgdc.rb]: These xpaths looks the same to me. Can it be one elsif clause with `institution.include?('MIT') || institution.include?('Tufts')`

File: lib/traject/macros/xml.rb (hunk 1 above)
Comment (jcoyne) [inline lib/traject/macros/xml.rb]: Coveralls is showing no coverage of this method, so it's probably not used.

File: lib/traject/macros/fgdc.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+        accumulator << identifier if identifier.present?
+      }
+    end
+
+    def select_identifier(record, context)
+      uuid = extract_uuid(record, context)
+      if uuid.present?
+        uuid
+      elsif context.settings.key?('command_line.filename')
```
Comment (jcoyne) [inline lib/traject/macros/fgdc.rb]: In which cases are these `elsif` clauses called?  They aren't tested.  Are they supporting real use cases?

File: lib/traject/fgdc_config.rb (hunk 3 above)
Comment (cbeer) [inline lib/traject/fgdc_config.rb]: (See #163 for more information about the config-driven approach)

Comment (cmh2166) [inline lib/traject/fgdc_config.rb]: yes, just seeing that. will do.

File: lib/traject/macros/fgdc.rb (hunk 4 above)
Comment (cmh2166) [reply lib/traject/macros/fgdc.rb]: done, thanks for that catch.

File: lib/traject/macros/fgdc.rb (hunk 5 above)
Comment (cmh2166) [reply lib/traject/macros/fgdc.rb]: the goal was to be a bit defensive if we ever get more than 4 fgdc records and we can't find ids. will add a test for it though.

**Pull request merged**